src/types.rs

```diff
@@ -10,6 +11,21 @@ use ethereum_consensus::{
 use serde::{de::DeserializeOwned, Deserialize, Serialize};
 use std::{collections::HashMap, fmt};
 
+#[derive(Parser, Default, Debug)]
+#[clap(version, about = "Beacon API client")]
+pub struct CliArgs {
```

Comment:
we can have a conversation about what the UI should look like, but what I had in mind was something like:

```bash
beacon-api-client --endpoint ENDPOINT $NAMESPACE $METHOD $ARGS
```

where `$NAMESPACE` maps to the high-level groupings in the docs
and `$METHOD` would be a compact description of what the method does

what does "compact" mean -- I tried to make a translation when coming up with the function names on the `Client` type, and you could even start here e.g. `beacon-api-client --endpoint $E beacon get_validators` etc

I bring this up to call out the "subcommand" functionality in `clap`; you can see an example here https://github.com/ralexstokes/mev-rs/blob/main/mev-rs/src/main.rs#L33

and again -- not required for now if you want to return to this point but I'll bring it up so you are aware; feel free to stick to something simpler while you are still doing more raw development

Reply:
one reason to do this is that we can then segregate the "argument" parsing logic to a handler of sorts for each method; e.g. something things will want a root that we can just supply (`beacon-api-client beacon block 0x....`) where the last bit is parsed as a `Root` type to supply to `Client::get_beacon_block`

if we do all of this in this way, the main routine for the CLI is essentially:
- match on namespace
- match on method we recognize
- parse args
- _if_ all of this succeeds (well-defined and well-formatted methods/args), then we can make a `Client` and call the appropriate endpoint